modular_skyrat/modules/clock_cult/code/research/clockwork_research.dm

```diff
@@ -0,0 +1,108 @@
+/datum/clockwork_research
+	/// Name of the research node
+	var/name = ""
+	/// Desc of the research
+	var/desc = ""
+	/// Extra fluff text for your research, moreso lore bits
+	var/lore = ""
+	/// List of tinker cache items it unlocks
+	var/list/unlocked_recipes = list()
+	/// List of scriptures this unlocks
+	var/list/unlocked_scriptures = list()
+	/// What tier this is
+	var/tier = 1
+	/// If this is a starting research
+	var/starting = FALSE
+	/// If this has been researched
+	var/researched = FALSE
+	/// How long this research's ritual will take to complete
+	var/time_to_research = 6 MINUTES
+	/// Valid areas where a research ritual may happen
+	var/static/list/area_whitelist = typecacheof(list(
+		/area/station/cargo,
+		/area/station/command,
+		/area/station/commons,
+		/area/station/construction,
+		/area/station/engineering,
+		/area/station/maintenance/disposal,
+		/area/station/maintenance/radshelter,
+		/area/station/maintenance/tram,
+		/area/station/medical,
+		/area/station/science,
+		/area/station/security,
+		/area/station/service,
+	))
+	/// Areas where you can't be tasked to start a ritual.
+	var/static/list/area_blacklist = typecacheof(list(
+		/area/station/engineering/supermatter,
+		/area/station/engineering/transit_tube,
+		/area/station/science/ordnance/bomb,
+		/area/station/science/ordnance/burnchamber,
+		/area/station/science/ordnance/freezerchamber,
+		/area/station/science/server,
+		/area/station/security/prison/safe,
+	))
+	/// A typepath of the area that this research's ritual must be done in.
+	var/area/selected_area
+
+
+/datum/clockwork_research/New(set_area = TRUE)
+	. = ..()
+	if(!selected_area && set_area)
+		set_new_area()
+
+
+/// Pick an an area for this research to take place in
+/datum/clockwork_research/proc/set_new_area()
+	var/list/possible_areas = GLOB.the_station_areas.Copy()
+	for(var/area/possible_area as anything in possible_areas)
+		if(initial(possible_area.outdoors) || !is_type_in_typecache(possible_area, area_whitelist) || is_type_in_typecache(possible_area, area_blacklist))
+			possible_areas -= possible_area
+
+	selected_area = pick(possible_areas)
+	validate_area()
+
+
+/// Makes sure the selected area is correct, and regenerates it if not. Potentially dangerous if there are no station areas, somehow.
+/datum/clockwork_research/proc/validate_area()
+	if(!selected_area || !length(get_area_turfs(selected_area)))
+		set_new_area()
+		return FALSE
+	return TRUE
+
+
+/// Makes sure the researching place is the correct area
```

Comment:
You may want to expand this autodoc comment, explaining when it returns `TRUE` and `FALSE`

Reply:
all changes done